model-optimizer/unit_tests/extensions/back/ShapeOfConstFolding_test.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import unittest
+
+import numpy as np
+
+from extensions.back.ShapeOfConstFolding import ShapeOfConstFolding
+from mo.front.common.partial_infer.eltwise import eltwise_infer
+from mo.front.common.partial_infer.utils import int64_array
+from mo.utils.ir_engine.compare_graphs import compare_graphs
+from unit_tests.utils.graph import build_graph
+
+const_value = np.random.rand(1, 3, 30, 30)
+nodes_attributes = {'input': {'shape': int64_array([1, 3, 30, 30]), 'type': 'Parameter', 'kind': 'op',
+                              'op': 'Parameter'},
+                    'input_data': {'value': None, 'shape': int64_array([1, 3, 30, 30]), 'kind': 'data'},
```

Comment:
nit: use helper functions for graph generation to create data nodes and connect edges.